src/ucp/core/ucp_worker.c

```diff
@@ -718,9 +718,13 @@ static void ucp_worker_iface_deactivate(ucp_worker_iface_t *wiface, int force)
               force, wiface->activate_count, worker->num_active_ifaces);
 
     if (!force) {
-        ucs_assert(wiface->activate_count > 0);
-        if (--wiface->activate_count > 0) {
-            return; /* not completely deactivated yet */
+        ucs_assert(worker->context->config.ext.proto_enable ||
+                   wiface->activate_count > 0);
+        if ((wiface->activate_count == 0) || (--wiface->activate_count > 0)) {
```

Comment:
consider splitting this to two 'if'
it will make it easier to see the 2nd if condition has a side-effect